test/aio5gc/lib/tools/nasCoder.go

```diff
@@ -46,7 +46,8 @@ func Encode(ue *context.UEContext, msg *nas.Message) ([]byte, error) {
 	}
 
 	// Plain NAS message
-	if ue == nil || !ue.SecurityContextAvailable {
+
+	if ue == nil || !(ue.GetSecurityContext() != nil) {
```

Comment:
You should use your state machine instead of doing this kind of check. You should have a branch of states where you are sure that securityContext is available. The goal of having a state machine is that when you arrive to a given state, you are sure that some preconditions are valid, here that the SecurityContext is available.
Then you have an Encode for state SecurityContextAvailable and an when it is not the case. Same for Decode.